app/actions/views/channel.js

```diff
@@ -351,3 +361,51 @@ export function setChannelDisplayName(displayName) {
         displayName
     };
 }
+
+// Returns true if there are more posts to load
+export function increasePostVisibility(channelId, focusedPostId) {
+    return async (dispatch, getState) => {
+        const state = getState();
+        const {loadingPosts, postVisibility} = state.views.channel;
+        const currentPostVisibility = postVisibility[channelId] || 0;
+
+        if (loadingPosts[channelId]) {
+            return true;
+        }
+
+        dispatch(batchActions([
+            {
+                type: ViewTypes.LOADING_POSTS,
+                data: true,
+                channelId
+            }
+        ]));
+
+        const page = Math.floor(currentPostVisibility / POST_INCREASE_AMOUNT);
+
+        let posts;
+        if (focusedPostId) {
+            posts = await getPostsBefore(channelId, focusedPostId, page, POST_INCREASE_AMOUNT)(dispatch, getState);
```

Comment:
Yeah, can't happen everywhere, but yeah we can save this for when a refactor is needed.